spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/integration/StreamToTableJoinIntegrationTests.java

```diff
@@ -208,6 +197,142 @@ public void testStreamToTable() {
 			assertThat(actualClicksPerRegion).hasSameElementsAs(expectedClicksPerRegion);
 		}
+		finally {
+			consumer.close();
+		}
+	}
+
+	@Test
+	public void testGlobalStartOffsetWithLatestAndIndividualBindingWthEarliest() throws Exception {
+		SpringApplication app = new SpringApplication(CountClicksPerRegionApplication.class);
+		app.setWebApplicationType(WebApplicationType.NONE);
+
+		Consumer<String, Long> consumer;
+		Map<String, Object> consumerProps = KafkaTestUtils.consumerProps("group-2", "false", embeddedKafka);
+		consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+		consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+		consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class);
+		DefaultKafkaConsumerFactory<String, Long> cf = new DefaultKafkaConsumerFactory<>(consumerProps);
+		consumer = cf.createConsumer();
+		embeddedKafka.consumeFromAnEmbeddedTopic(consumer, "output-topic-2");
+
+		// Produce data first to the input topic to test the startOffset setting on the
+		// binding (which is set to earliest below).
+		// Input 1: Clicks per user (multiple records allowed per user).
+		List<KeyValue<String, Long>> userClicks = Arrays.asList(
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L),
+				new KeyValue<>("alice", 100L)
+		);
+
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		senderProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+		senderProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, LongSerializer.class);
+
+		DefaultKafkaProducerFactory<String, Long> pf = new DefaultKafkaProducerFactory<>(senderProps);
+		KafkaTemplate<String, Long> template = new KafkaTemplate<>(pf, true);
+		template.setDefaultTopic("user-clicks-2");
+
+		for (KeyValue<String, Long> keyValue : userClicks) {
+			template.sendDefault(keyValue.key, keyValue.value);
+		}
+		//Thread.sleep(10000L);
+		try (ConfigurableApplicationContext ignored = app.run("--server.port=0",
+				"--spring.jmx.enabled=false",
+				"--spring.cloud.stream.bindings.input.destination=user-clicks-2",
+				"--spring.cloud.stream.bindings.input-x.destination=user-regions-2",
+				"--spring.cloud.stream.bindings.output.destination=output-topic-2",
+				"--spring.cloud.stream.bindings.input.consumer.useNativeDecoding=true",
+				"--spring.cloud.stream.bindings.input-x.consumer.useNativeDecoding=true",
+				"--spring.cloud.stream.bindings.output.producer.useNativeEncoding=true",
+				"--spring.cloud.stream.kafka.streams.binder.configuration.auto.offset.reset=latest",
+				"--spring.cloud.stream.kafka.streams.bindings.input.consumer.resetOffsets=true",
```

Comment:
I don't see this property being used anywhere in the streams binder; the message channel binder seeks to beginning/end when this is true.

Reply:
Removed it. This was a leftover and an oversight from my side to not cleanup.

Reply:
Hehe - my question is really "should it be honored?", or "should we detect it and reject the configuration since we don't honor it" 😄

Reply:
Honoring `resetOffset` in the streams binder is going to be tricky. See [this discussion](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/376#issuecomment-388207759). This is because of all the intermediate changelog topics kafka streams create behind the scenes. So I think it is like a no-op configuration. If the user provides it, that has no effect. What do you mean by "detect and reject"? In effect, thats the end result now although the binder doesn't do any detection of that property.

Reply:
Well, I don't think you would want to mess with the intermediate topics, just the input topic. But, I agree, it's messy if the intermediates aren't reset to the end when the input is reset (to either end).

It's your call, but throwing an exception if it's configured, but not used, might avoid user confusion. Or, at least add a note to the docs.